core/src/main/java/org/eclipse/hono/util/ProtocolAdapterMessageHelper.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * Utility methods for working with messages that arrive at a protocol adapter.
+ * <p>
+ * Methods contained here are provided for the protocol adapters and do not cover any Proton {@code Message}s.
+ * See {@link MessageHelper} for utility methods that are related to these kind of messages.
+ *
+ */
+public final class ProtocolAdapterMessageHelper {
```

Comment:
we already have a place for utility methods used by all protocol adapters: `AbstractProtocolAdapterBase`

why do we need to introduce an additional place?

Reply:
You are right, this fitted well into the `AbstractProtocolAdapterBase`, will eliminate the helper class.